internal/output/disk/disk.go

```diff
@@ -29,33 +31,40 @@ func (d *Disk) TransactionChan() chan<- types.Transaction {
 
 //Init func Initialize the disk output plugin
 func (d *Disk) Init(config types.Config, logger zap.Logger) error {
-	d.FileName = config["filename"].(string)
-	d.FilePath = config["filepath"].(string)
+	if d.FilePath,d.TypeCheck = config["filepath"].(string); !d.TypeCheck {
+		return errors.New("FilePath must be a string")
+	}
 	path.Clean(d.FilePath)
-	d.FilePermission = config["permission"].(os.FileMode)
-	d.Transactions = make(chan types.Transaction, 1)
+	if d.FilePermission,d.TypeCheck = config["permission"].(string); !d.TypeCheck{
+		return errors.New("FilePermission must be from a string")
+	}
+	d.Transactions = make(chan types.Transaction)
 	d.stopChan = make(chan struct{})
 	d.logger = logger
 	return nil
 }
 
-//WriteOnDisk func takes the disk and the transaction
+//WriteOnDisk func takes the transaction
 //that to be written on the disk
-func WriteOnDisk(d *Disk, transaction types.Transaction) {
+func (d *Disk) WriteOnDisk(transaction types.Transaction) {
 	defer d.wg.Done()
-	var err error
 	ack := true
-	d.logger.Info("RECEIVED OUTPUT TRANSACTION...")
-	bytes, _ := ioutil.ReadAll(transaction)
-	if _, err = os.Stat(d.FilePath); os.IsNotExist(err) {
-		err = os.MkdirAll(d.FilePath, os.ModePerm)
+	bytes, err := ioutil.ReadAll(transaction)
```

Comment:
You can read bytes only when you're sure you'll write, consider moving the reading part only when you're about to write.